### PR TITLE
replace deprecated CURLOPT_PROTOCOLS with CURLOPT_PROTOCOLS_STR

### PR DESCRIPTION
**Pull request by babaissarkar**
According to [Curl Docs](https://curl.se/libcurl/c/CURLOPT_PROTOCOLS.html), CURLOPT_PROTOCOLS is deprecated with `curl_easy_setopt()`.

### CODE REVIEW

**Comment by Pentarctagon** [comment]
Note that this was added in curl 7.85 whereas Ubuntu 22.04 currently only has 7.81. So I would expect the CI to fail to build.

**Comment by babaissarkar** [comment]
> Note that this was added in curl 7.85 whereas Ubuntu 22.04 currently only has 7.81. So I would expect the CI to fail to build.

Does this mean that it can't be merged yet?

**Comment by Pentarctagon** [comment]
Correct.

**Comment by soliton-** [comment]
Presumably you can check some pre-processor symbol to decide the curl version and write conditional code.

**Comment by babaissarkar** [comment]
> Presumably you can check some pre-processor symbol to decide the curl version and write conditional code.

I only managed to find this : https://curl.se/libcurl/c/curl_version_info.html

**Comment by Pentarctagon** [comment]
I plan to update the CI to Ubuntu 24.04 once Mint 22 is released (probably in a few months) since that'll be based on Ubuntu 24.04 so I'll be able to locally troubleshoot Linux CI issues. So that'd be the earliest this could go in without adding some sort of conditional compilation. Also given this would make it a requirement to have at least a specific, more recent version of curl, that info should be added to INSTALL.md with this PR as well.

Alternatively, you could add in ifdef based on:
https://github.com/curl/curl/blob/b9c2a56be2980c1d2f6192006ac5d306bbb6858b/include/curl/curlver.h#L39-L41

**Comment by soliton-** [comment]
IMO a condition on the version is very much preferred instead of forcing everyone to upgrade at the same time.

File: src/network_download_file.cpp [hunk 1]
```diff
@@ -53,7 +53,11 @@ namespace network
             curl_easy_setopt(curl, CURLOPT_FRESH_CONNECT, 1L);
             curl_easy_setopt(curl, CURLOPT_FAILONERROR, 1L);
             curl_easy_setopt(curl, CURLOPT_TIMEOUT_MS, 5000L);
-            curl_easy_setopt(curl, CURLOPT_PROTOCOLS_STR, CURLPROTO_HTTPS);
+            #if LIBCURL_VERSION_NUM > 0x075500
```
Comment (soliton-) [inline src/network_download_file.cpp:56]: Should be `>=`, no?

Comment (babaissarkar) [reply src/network_download_file.cpp]: Done.

**Review by soliton- — APPROVED** [review]
Should be squashed.

File: src/network_download_file.cpp [hunk 2]
```diff
@@ -53,7 +53,11 @@ namespace network
             curl_easy_setopt(curl, CURLOPT_FRESH_CONNECT, 1L);
             curl_easy_setopt(curl, CURLOPT_FAILONERROR, 1L);
             curl_easy_setopt(curl, CURLOPT_TIMEOUT_MS, 5000L);
-            curl_easy_setopt(curl, CURLOPT_PROTOCOLS, CURLPROTO_HTTPS);
+            #if LIBCURL_VERSION_NUM >= 0x075500
```
Comment (stevecotton) [inline src/network_download_file.cpp:56]: Is that the right number? We need 7.85 or later, and that looks like 7.55.

Comment (babaissarkar) [reply src/network_download_file.cpp:56]: `LIBCURL_VERSION_NUM` is a hexadecimal number, see https://github.com/curl/curl/blob/b9c2a56be2980c1d2f6192006ac5d306bbb6858b/include/curl/curlver.h#L54.

Comment (stevecotton) [reply src/network_download_file.cpp:56]: OK, sorry for raising it.

**Comment by stevecotton** [comment]
Given that it includes conditional compilation, I'm intending to backport to 1.18; however in doing so I've noticed that it needs a tabs vs spaces cleanup.

**Comment by babaissarkar** [comment]
> Given that it includes conditional compilation, I'm intending to backport to 1.18; however in doing so I've noticed that it needs a tabs vs spaces cleanup.

Any problems if I commit the cleaned up version directly?

**Comment by stevecotton** [comment]
I prefer everything through CI, but yeah, it's trivial, so either is okay.

**Comment by babaissarkar** [comment]
> I prefer everything through CI, but yeah, it's trivial, so either is okay.

NVM, I opened a PR. See #8999.

**Comment by CelticMinstrel** [comment]
There's no reason to run CI on something like that – even if it's a PR it probably should put `[ci skip]` in the commit message.

**Comment by stevecotton** [comment]
CI on everything means a little more work for one person on that change, but saves work for a lot of people on the occasions when the CI catches something.

**Pull request merged**